app/controllers/items_controller.rb

```diff
@@ -213,8 +213,11 @@ def refresh_metadata
 
   # This is called from the item page and from the bulk (synchronous) update page
   def set_rights
-    @object.read_rights = params[:access_form][:rights]
-    save_and_reindex
+    object_client = Dor::Services::Client.object(@object.pid)
+    dro = object_client.find
+    json = JSON.parse(dro.to_json)
```

Comment:
Instead of `JSON.parse(dro.to_json)`, can you do `dro.to_h`? Isn't the thing in the `json` variable a hash anyway? I guess I'm wondering: why bother with the conversion to and then immediately from JSON if we have a cocina model that might/should support `to_h`?

Reply:
No, because dro.to_h will return date objects and then if you send that to update it causes an error because the dates are not represented as strings.  I think we can look into how we improve this interaction, but I didn't want to get into it here.

Reply:
OK, sounds good.